src/go/rpk/pkg/cli/cmd/container/start_test.go

```diff
@@ -508,3 +515,39 @@ Please check your internet connection and try again.`,
 		})
 	}
 }
+
+func TestCollectFlags(t *testing.T) {
+	tests := []struct {
```

Comment:
i think we need a failing test

```
--set "foo=bar" "baz" "welp"
```
baz and welp should fail or alternatively it sould try to set `baz=welp` one of the two options, but should be well defined.

Reply:
In this case, `"baz"` & `welp` would get filtered out, as seen in the tests: https://github.com/vectorizedio/redpanda/pull/1425/commits/a9877e625d6955ff9d272ca93ec162dd780df0cb#diff-447f1d1e4b0a771d0a8a20761614f1f1dbd4f9c501b723f3efef10c0a61550cdR524-R544

The thing is that there are some flags, such as `--image`, `--nodes` or `--verbose` which are parsed by `Cobra`. So we can't fail upon encountering a flag that isn't `--set` because it might be one of them.

Reply:
Makes sense! Thank you